pkg/metrics/metrics.jsx

```diff
@@ -489,64 +512,146 @@ class CurrentMetrics extends React.Component {
         );
 
         const topServicesMemory = n_biggest(this.cgroupMemoryNames, this.samples[10], 5);
         newState.topServicesMemory = topServicesMemory.map(
-            ([key, value, is_user, is_container, userid]) => cgroupRow(key, cockpit.format_bytes(value), is_user, is_container, userid)
+            ([key, value, is_user, is_container, userid]) => this.cgroupRow(key, is_user, is_container, userid, cockpit.format_bytes(value))
         );
 
-        const notMappedContainers = topServicesMemory.concat(topServicesCPU).filter(([key, value, is_user, is_container, userid]) => is_container && getCachedPodName(userid, key) === undefined);
+        const notMappedContainers = topServicesMemory.concat(topServicesCPU).filter(([key, value, is_user, is_container, userid]) => is_container && this.getCachedPodName(userid, key) === undefined);
         if (notMappedContainers.length !== 0) {
             this.update_podman_name_mapping(notMappedContainers);
         }
         this.setState(newState);
     }
 
+    onPrivilegedMetricsUpdate(event, message) {
+        debug("process metrics message", message);
+        const data = JSON.parse(message);
+
+        if (!Array.isArray(data)) {
+            this.cgroupDiskNames = data.metrics[0].instances.slice();
+            return;
+        }
+
+        data.forEach(privilegedSamples => decompress_samples(privilegedSamples, this.privilegedSamples));
+
+        // return [ name, read, write, isUser, isContainer, uid | cgroup ]
+        const n_biggest = (n, names, valuesA, valuesB) => {
+            const merged = [];
+            const userSlices = {};
+            names.forEach((name, i) => {
+                // filter out invalid values, the empty (root) cgroup, non-services
+                if (name.endsWith('.service')) {
+                    // only keep cgroup basenames, and drop redundant .service suffix
+                    const label = name.replace(/.*\//, '').replace(/\.service$/, '');
+                    const isUser = name.match(useridCgroupRe);
+                    merged.push([label, valuesA[i], valuesB[i], isUser, false, name]);
+                    return;
+                }
+
+                // filter out podman containers, but only for the logged in
+                // user or root user if the user is privileged. Other users
+                // containers will show up under the user@uid cgroup
+                const matches = name.match(/libpod-(?<containerid>[a-z|0-9]{64})\.scope/);
+                if (matches && valuesA[i] !== undefined && valuesB[i] !== undefined) {
+                    let isUser = false;
+                    let uid = 0;
+                    const containerid = matches.groups.containerid;
+                    const umatches = name.match(useridCgroupRe);
+                    if (umatches) {
```

Comment:
This could be nicer riten like:
const isUser = umatches;
const uid = parseInt(umtches?.groups.userid) || 0;